google-apis-core/lib/google/apis/core/http_command.rb

```diff
@@ -355,54 +355,42 @@ def safe_response_representation http_res
           http_res.inspect
         end
 
-        def opencensus_begin_span
-          return unless OPENCENSUS_AVAILABLE && options.use_opencensus
-          return if @opencensus_span
-          return unless OpenCensus::Trace.span_context
-
-          @opencensus_span = OpenCensus::Trace.start_span url.path.to_s
-          @opencensus_span.kind = OpenCensus::Trace::SpanBuilder::CLIENT
-          @opencensus_span.put_attribute "http.host", url.host.to_s
-          @opencensus_span.put_attribute "http.method", method.to_s.upcase
-          @opencensus_span.put_attribute "http.path", url.path.to_s
-          if body.respond_to? :bytesize
-            @opencensus_span.put_message_event \
-              OpenCensus::Trace::SpanBuilder::SENT, 1, body.bytesize
-          end
+        def opentelemetry_begin_span
+          return unless OPENTELEMETRY_AVAILABLE && Google::Apis::Core::Instrumentation.instance.installed?
+          return if @opentelemetry_span
 
-          formatter = OpenCensus::Trace.config.http_formatter
-          if formatter.respond_to? :header_name
-            header[formatter.header_name] = formatter.serialize @opencensus_span.context.trace_context
-          end
+          attributes = {
+            'http.host' => url.host.to_s,
+            'http.method' => method.to_s,
+            'http.target' => url.path.to_s,
+            'peer.service' => 'google'
```

Comment:
I don't think most google backends are actually emitting spans or setting `service.name` so it may not matter that much... But from my understanding of the semantics of this attribute, I suggest using `url.host.to_s` which will return the "official" ID of the service being called.